edk2toolext/tests/test_nuget_dependency.py

```diff
@@ -110,7 +110,7 @@ def test_download_bad_nuget(self):
 
         ext_dep_descriptor = EDF.ExternDepDescriptor(ext_dep_file_path).descriptor_contents
         ext_dep = NugetDependency(ext_dep_descriptor)
-        with self.assertRaises(RuntimeError):
+        with self.assertRaises(ValueError):
```

Comment:
I don't see why this would raise a ValueError?

Reply:
We only normalize the version when we fetch from cache

Reply:
My guess is here: https://github.com/tianocore/edk2-pytool-extensions/blob/4d881f5f4bf13baffa444eabc258cf5723e690da/edk2toolext/environment/extdeptypes/nuget_dependency.py#L95

Reply:
If we don't find a global cache, we bail. My guess is the server doesn't have one. So it bails and blows up with a RuntimeError later in fetch. Your local build has a cache and blows up earlier when it tries to normalize the version. I think we could either normalize the version earlier, or catch either exception.